federation-js/src/index.ts

```diff
@@ -1,8 +1,8 @@
 import { buildSubgraphSchema } from '@apollo/subgraph';
```

Comment:
If this is all that's left of the exports, should we just remove the package and stop publishing it? Its 2 purposes (subgraph and composition) are both completely owned by separate packages, seems reasonable to remove this altogether going into v2.

Reply:
This comment is directed at the whole file, not just the one line.

Reply:
> seems reasonable to remove this altogether going into v2

I would agree (and double-checking, it doesn't seem the module is used anymore within federation itself after this PR). I think the only concerns is that many people will still have imports on `@apollo/federation` for `buildSubgraphSchema` and so the question is whether we want to maintain compatibility for that. But as fed 2 is a major, I'd personally say it's a good time for that (if anything, stopping to publish the package would be a good sign that people should move to use `@apollo/subgraph`).

The type I felt was maybe imported by users (and that the PR moves directly into `federation-js/src/index.ts` for that reason) is the `ServiceDefinition` interface because it's used for local gateway configs. But it would arguably make more sense that this interface is exported by `@apollo/gateway` so we could make it so.

@abernix Opinions?

Reply:
Agree with all of the above! v2 seems like exactly the time to remove deprecated bits like this, especially if it means we can stop publishing a package altogether. Also agree with the `ServiceDefinition` export finding a new home in `@apollo/gateway`, that sgtm.

Reply:
@trevor-scheer @pcmanus I support this, but it would be worth opening an issue that ensures that we at some point run `npm deprecate <pkg> <msg>` on this package.  Doing that will start outputting the deprecation warnings during installation of the package by package managers and be a good signal to developers to make the (remarkably straightforward) change to the new package name. We might even want to do this quite soon to get the gears in motion?

It's worth calling out the impact this has on tutorials and education content that we and others have produced — and even printed books!  Those things can all be updated, but in my mind the sooner we get the deprecation warning up and the longer we keep it working (perhaps by having `@apollo/federation` depend on a caret semver range that automatically gets minor/patch updates to `@apollo/subgraph` for some time?).  We may even want to communicate this directly to those that are thinking about developer experience / docs / blogs / etc.

Reply:
I think deprecating only applies to v2 (which we maybe just shouldn't have ever published given that we're open to killing it). The v0.x gateway continues to use `@apollo/federation` for its composition capabilities, so I don't think we want to deprecate it (unless we broke out composition as well, which I'm not sure is worth the effort).

Reply:
Hmm, I see what you're saying.  Perhaps what we might consider is then adding a runtime deprecation warning to the use of `@apollo/federation` for its `buildFederatedSchema` / `buildSubgraphSchema`, rather than at the npm level?

Reply:
I think we want to:
* `npm deprecate` v2+ versions of `@apollo/federation`
* Runtime deprecate `buildSubgraphSchema` (currently re-exported from `@apollo/subgraph`)

Currently `buildFederatedSchema` is already runtime deprecated in `@apollo/federation` and isn't exported by `@apollo/subgraph` at all.